Quaver.Shared/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -837,5 +876,26 @@ public override UserClientStatus GetClientStatus()
 
             return new UserClientStatus(status, Map.MapId, MapHash, (byte) Ruleset.Mode, content, (long) ModManager.Mods);
         }
+
+        /// <summary>
+        /// </summary>
+        public void HandleSuggestedOffsetCalculations()
+        {
+            var stats = Ruleset.ScoreProcessor.Stats.FindAll(x => x.KeyPressType == KeyPressType.Press);
+
+            var offset = 0;
+
+            // Calculate new offset
+            if (stats.Count != 0)
+            {
+                var samples = new List<double>(stats.Select(x => (double) x.HitDifference).ToList());
+                var stdDev = samples.StandardDeviation();
+
+                if (Math.Abs(stdDev) < Ruleset.ScoreProcessor.JudgementWindow[Judgement.Good])
+                    offset = (int) samples.Average() + ConfigManager.GlobalAudioOffset.Value;
+            }
+
+            DialogManager.Show(new OffsetConfirmDialog(this, offset));
```

Comment:
Should be inside both ifs I think.

Reply:
The dialog is the way to exit the screen, so it should always display.

Reply:
It should not suggest offset = 0 because that's an error case.

Reply:
Auto-restart then?

Reply:
If I entered the calibrator by accident, I'll escape out asap without pressing anything, in which case I'd like to quit without any confirmation.

Reply:
Made it so it just exits with a notification if it can't calculate it. You can still exit normally w/ escape at anytime.